File: youlexuan-shop-web/src/main/webapp/js/service/typeTemplateService.js
app.service('typeTemplateService',function ($http) {

    this.delete = function (ids) {
        return $http.get('../typeTemplate/delete.do?ids='+ids)
    }

    this.findOne = function (id) {
        return $http.get('../typeTemplate/findOne.do?id='+id);
    }

    this.findPage = function (pageNum, pageSize, searchEntity) {
        return $http.post('../typeTemplate/findPage.do?pageNum='+pageNum+'&pageSize='+pageSize,searchEntity);
    }
    
    this.initData = function () {
        return $http.get('../typeTemplate/initData.do');
    }

    this.add = function (entity) {
        return $http.post('../typeTemplate/add.do',entity);
    }

    this.update = function (entity) {
        return $http.post('../typeTemplate/update.do',entity);
    }

});

File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/TypeTemplateService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbTypeTemplate;

import java.util.Map;

public interface TypeTemplateService {
    public PageResult findPage(int pageNum,int pageSize,TbTypeTemplate typeTemplate);

    /**
     * 查询 品牌和规格  的两个集合，存入到同一个map中
     * @return
     */
    public Map initAddTypeTempateData();

    /**
     *添加模板
     * @param typeTemplate
     */
    public void add(TbTypeTemplate typeTemplate);

    /**
     * 更新
     * @param typeTemplate
     */
    public void update(TbTypeTemplate typeTemplate);

    /**
     * 根据id主键查询
     * @param id
     * @return
     */
    public TbTypeTemplate findOne(Long id);

    /**
     * 批量删除
     * @param ids
     */
    public void delete(Long[] ids);
}
